### PR TITLE
Add a saltcloud driver for Hetzner Cloud

### PR DESCRIPTION
**Pull request by dithmer**
### What does this PR do?
This PR adds a new saltcloud driver for a german settled provider we are using, called Hetzner GmbH. Hetzner GmbH provides a public cloud service (Hetzner Cloud - https://hetzner.cloud) with a public api and a maintained Python client/library which is used in this driver.

It also adds a example configuration for a provider and profile, as well as the stubs for integration tests. We were using the linode driver as an example and tried to do things mostly like the linode driver does, as told in the saltcloud development docs.

It is known that there are still the docs missing. We are working on that, but hoped that the code review can already be started.

Hetzner allows projects in open source context to have a test account with a balance of 50,-€. If desired for testing I would request such an account.

### What issues does this PR fix or reference?
-

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated (Stubs, as done in the linode driver which is referenced in the salt-cloud development docs https://docs.saltstack.com/en/latest/topics/cloud/cloud.html#the-create-function)

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by LKaemmerling** [comment]
Hey @dithmer,

thank you for this integration into salt! We (Hetzner Cloud) are really happy about every integration the community performs. I just found one small thing, you refer to the Hetzner Cloud as "Hetzner Public Cloud". It would be nicer to just call ist "Hetzner Cloud", as there is no "Hetzner Public (or Private) Cloud". Even within the whole Hetzner Cloud community, the Hetzner Cloud is known as  `hcloud`. If you would need help with the testing, just write me or fill out a Support Ticket via the Cloud Console, we might help here with credits :)

Thank you!
 - Lukas
Hetzner Cloud

**Comment by dithmer** [comment]
> Hey @dithmer,
> 
> thank you for this integration into salt! We (Hetzner Cloud) are really happy about every integration the community performs. I just found one small thing, you refer to the Hetzner Cloud as "Hetzner Public Cloud". It would be nicer to just call ist "Hetzner Cloud", as there is no "Hetzner Public (or Private) Cloud". Even within the whole Hetzner Cloud community, the Hetzner Cloud is known as `hcloud`. If you would need help with the testing, just write me or fill out a Support Ticket via the Cloud Console, we might help here with credits :)
> 
> Thank you!
> 
> * Lukas
>   Hetzner Cloud

Hey @LKaemmerling

Thanks for providing the new version of the hcloud python library for Python 3.9 that quick! Really appreciate that! As you may have seen, I already fixed the naming (Hetzner Public Cloud -> Hetzner Cloud) whereever I could find it! Thank you very much for that advise :) 

Thanks as well for the kind offer of help/support/credits. I will message you or write a ticket if needed!

Best Regards!

File: tests/integration/cloud/clouds/test_hcloud.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+"""
+    Tests for the Hetzner-Public-Cloud driver
```
Comment (LKaemmerling) [inline tests/integration/cloud/clouds/test_hcloud.py:3]: @dithmer i just found that here is still refered to "Public-Cloud" (and below too)

**Comment by dithmer** [comment]
@xeacott Is there anything we can or have to do to make this happen? :-)

**Pull request closed without merging**